qiskit_machine_learning/neural_networks/sampler_qnn.py

```diff
@@ -82,35 +95,40 @@ def __init__(
         Raises:
             QiskitMachineLearningError: Invalid parameter values.
         """
-        # set primitive
+        # set primitive, provide default
+        if sampler is None:
+            sampler = Sampler()
         self.sampler = sampler
 
         # set gradient
-        # TODO: provide default gradient?
-        self.gradient = gradient or ParamShiftSamplerGradient(self.sampler)
+        if gradient is None:
+            gradient = ParamShiftSamplerGradient(self.sampler)
```

Comment:
The default ought to be to noted in the arg text above where its states it as optional sampler gradient...